extensions/hazelcast-client/README.MD

```diff
@@ -0,0 +1,62 @@
+# Hazelcast Client for Quarkus
+
+[Hazelcast IMDG](https://hazelcast.com/products/imdg/) is a distributed in-memory object store and compute engine 
+that supports a wide variety of data structures such as Map, Set, List, MultiMap, RingBuffer, HyperLogLog. 
+
+Use Hazelcast IMDG to store your data in RAM, spread and replicate it across a cluster of machines, 
+and perform data-local computation on it. 
+
+Hazelcast is:
+- cloud and Kubernetes friendly
+- often used as a Distributed Cache amongst other use cases
+
+## Overview
+
+Extension exposes a single native-mode-compatible Hazelcast Client bean (`HazelcastInstance`) which can be directly injected into your beans:
+```
+@Inject
+HazelcastInstance hazelcastClient;
+```
+
+## Quarkus hazelcast-client configuration
+
+Default configuration can be customized using `application.properties` entries such as:
+
+    quarkus.hazelcast-client.cluster-members
+    quarkus.hazelcast-client.outbound-port-definitions
+    quarkus.hazelcast-client.outbound-ports
+    quarkus.hazelcast-client.labels
+    quarkus.hazelcast-client.group-name
+    quarkus.hazelcast-client.connection-attempt-limit
+    quarkus.hazelcast-client.connection-attempt-period
+    quarkus.hazelcast-client.connection-timeout
+    quarkus.hazelcast-client.executor-pool-size
+
+If you need more configuration options, provide your own `hazelcast-client.yaml/hazelcast-client.xml` file, 
+which gives you full-blow configuration possibilities.
+
+If your use case is more demanding, you can wire `HazelcastInstance` bean yourself - keep in mind that you will be able to benefit from GraalVM compatibility anyway!
+
+## Limitations (native mode)
+- Default Java serialization is not supported
+- User code deployment is not supported
+- SSL support requires extra runtime configuration
+
+### SSL
+In order for SSL to work, one needs to properly set `java.library.path` and `javax.net.ssl.trustStore` runtime parameters.
+
+Example Dockerfile:
+
+```
+FROM quay.io/quarkus/ubi-quarkus-native-image:19.2.1 as graalvm
```

Comment:
This is outdated as we now only support 19.3.1 in master and that's my problem with having a README here: it will quickly be outdated. All this should be in the guide we publish to the website instead IMHO. We have attributes there and they are probably updated for each release.

Reply:
I agree, I know already that there's a separate place for guides and I'm planning to move it all there!